### ISSUE TITLE
Code Tests - Presentation stage

### ISSUE THREAD

**Issue by alonshmilo** [alonshmilo/MedicalData_jce#25]
# Issue: Code Tests - Presentation stage

### Explanation:
Code Tests, borderline cases.

### Checklist:
- [ ] Code
- [ ] Reading
- [ ] Documenting reading
- [ ] Project book
- [ ] Advisor consult

**Comment by stavdv** [comment]
Need to get things organized: 
Deepmedic: Synthetic data, dice and accuracy.
Yolo: coco data and our data on several images - high and low resoloution.

**Comment by stavdv** [comment]
Preliminary checks Deep medic:
* tinyCnn wtih syntetic data.
* path Check in the configuration files.
* Validation of image dimensions in cddHelpers (line 106)
* Usage checks (in ./deepMedicRun line 100)
  - [-newModel] and [-model] options cannot be used together.
  - [-test] cannot be used in conjuction with [-newModel] or [-train].
  - The option [-pretrained] can only be used together with the [-newModel] option:
  - The option [-layers] can only be used together with the [-pretrained] option.
  - The option [-resetOptimizer] can only be used together with the [-train] option.
* Accuracy monitoring per classes for subepoch - (in accurcyMonitor.py)
  - Mean accuracy: (TruePos+TrueNeg)/All predicted voxels.
  - Mean sensitivity: TruePos/RealPos
  - Mean specificity: TrueNeg/RealNeg
  - Mean Dice:[ 2 * (All common between TruePos and  RealPos)]/ TruePos + RealPos
* In cnn3d.py:
  - check Mean Validation Accuracy Of Last Epoch And Update Cnns Top Accuracy Achieved If 
     Needed,Called at the end of an epoch, right before increasing self.numberOfEpochsTrained.
  - check Training State Attributes Initialized
* Learning Rate Schedule. (in trainValidate...)
  - check the flag that should have been set True from the start.
    if training should do Auto-schedule. If we get in here, this is a bug.
  - dimensions check

**Comment by alonshmilo** [comment]
Yolo checks:
- GT image with original weights
- Scan image with original weights.
- Scan image with our trained weights.

**Comment by alonshmilo** [comment]
👍
